### PR TITLE
LLVM support for function pointers 

### LINKED ISSUE #523: LLVM support for function pointers
In order to do #402, I want to be able to create global constants that are structures that include function pointers.

This means we need to be able to create a function type and get a pointer to a function as a ConstPointerValue.

### PR DESCRIPTION
**Pull request by heshanpadmasiri**
## Purpose
LLVM support for function pointers 
Resolves #523 

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: compiler/modules/print.llvm/tests/fnTy.bal [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+function fnTy() returns Module {
+    Context context = new;
+    Builder builder = context.createBuilder();
+    Module m = context.createModule();
+    FunctionType fnTy = functionType("i64", ["i8", "i64"]);
+    PointerType fnPointerTy = pointerType(fnTy, 0);
+    StructType structTy = structType(["i32", fnPointerTy]);
+    ConstPointerValue g1 = m.addGlobal(structTy, "g1");
+    ConstPointerValue g2 = m.addGlobal(fnPointerTy, "g2");
```
Comment (jclark) [inline compiler/modules/print.llvm/tests/fnTy.bal:14]: I also need to be able to have an initializer that initializes it to a pointer to function that I have declared.

Comment (heshanpadmasiri) [reply compiler/modules/print.llvm/tests/fnTy.bal:14]: fixed

File: compiler/modules/print.llvm/common.bal [hunk 2]
```diff
@@ -61,6 +61,10 @@ public type FunctionType readonly & record {|
     Type[] paramTypes;
 |};
 
+public function functionType(RetType returnType, Type[] paramTypes) returns FunctionType {
+    return { returnType, paramTypes: paramTypes.cloneReadOnly()};
```
Comment (KavinduZoysa) [inline compiler/modules/print.llvm/common.bal]: space is needed at the end

**Review by jclark — APPROVED** [review]
Looks good

**Pull request merged**